src/impl_methods.rs

```diff
@@ -1719,7 +1723,8 @@ where
             let s = self.strides.axis(axis) as Ixs;
             let m = self.dim.axis(axis);
             if m != 0 {
-                self.ptr = self.ptr.offset(stride_offset(m - 1, s as Ix));
+                self.ptr =
+                    NonNull::new(self.ptr.as_ptr().offset(stride_offset(m - 1, s as Ix))).unwrap();
```

Comment:
We should probably avoid these round trips and have a NonNull version of the offset/strided offset.